src/sortby.ts

```diff
@@ -0,0 +1,95 @@
+/**
+  * Sort array by a specific element property, its value type must be one of string, number and date
+  * @param {string | undefined} sortByProperty specify a property from each element that sorting will be based on, undefined means sorty by element itself
+  * @param {
+      isDescending: boolean;
+      sortByPropertyType: string;
+    } options 
+  * @returns script version
+  */
+export function SortBy<T>(sortByProperty: string | symbol | number, options: {
+  isDescending: boolean;
+  type: string;
+} = {
+  isDescending: true,
+  type: 'string'
+}) {
+  const cachedValueKey = Symbol();
+  return function(
+    target: any,
+    propertyKey: string,
+    descriptor?: PropertyDescriptor
+  ) {
+    Object.defineProperty(target, propertyKey, {
+      set: function(value: Array<T>) {
+        
+        if (!value || !Array.isArray(value)) {
+          throw `Value of property ${propertyKey} is not a valid array!`;
+        }
+
+        if (!!sortByProperty && value.some(item => !item.hasOwnProperty(sortByProperty))) {
```

Comment:
I think we can skip this check if someone passed a `sortByProperty` it's the consumer's responsibility

Reply:
Also removed